Easydict/NewApp/View/SettingView/Tabs/DisabledAppTab.swift

```diff
@@ -0,0 +1,225 @@
+        }
+        .listStyle(.plain)
+        .scrollIndicators(.never)
+    }
+
+    var appListViewWithToolbar: some View {
+        VStack(spacing: 0) {
+            appListView
+
+            listToolbar
+        }
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+        .padding(.bottom)
+        .padding(.horizontal, 35)
+        .onTapGesture {
+            disabledAppViewModel.selectedAppModels = []
+        }
+    }
+
+    var body: some View {
+        VStack {
+            Text("disabled_title")
+                .padding()
+
+            appListViewWithToolbar
+        }
+        .frame(maxWidth: 500)
+        .environmentObject(disabledAppViewModel)
+        .onAppear {
+            disabledAppViewModel.fetchDisabledApps()
+        }
+    }
+}
+
+@available(macOS 13.0, *)
+struct ListToolbar: View {
+    @EnvironmentObject private var disabledAppViewModel: DisabledAppViewModel
+
+    var body: some View {
+        VStack(spacing: 0) {
+            Divider()
+            HStack(spacing: 0) {
+                ListButton(imageName: "plus") {
+                    disabledAppViewModel.isImporting.toggle()
+                }
+                Divider()
+                ListButton(imageName: "minus") {
+                    disabledAppViewModel.removeDisabledApp()
+                }
+                .disabled(disabledAppViewModel.selectedAppModels.isEmpty)
+                Spacer()
+            }
+            .padding(2)
+        }
+        .frame(height: 28)
+        .background(Color(.controlBackgroundColor))
+    }
+}
+
+@available(macOS 13.0, *)
+struct ListButton: View {
+    var imageName: String
+    var action: () -> Void
+
+    var body: some View {
+        Button(action: {
+            action()
+        }) {
+            Image(systemName: imageName)
+                .resizable()
+                .scaledToFit()
+                .frame(width: 12, height: 12)
+                .padding(6)
+                .contentShape(Rectangle())
+        }
+        .buttonStyle(BorderlessButtonStyle())
+    }
+}
+
+@available(macOS 13.0, *)
+struct BlockAppItemView: View {
+    @StateObject private var appFetcher: AppFetcher
+
+    @EnvironmentObject var disabledAppViewModel: DisabledAppViewModel
+
+    init(with appModel: EZAppModel) {
+        _appFetcher = StateObject(wrappedValue: AppFetcher(appModel: appModel))
+    }
+
+    var body: some View {
+        HStack(alignment: .center) {
+            Image(nsImage: appFetcher.appIcon ?? NSImage())
+                .resizable()
+                .scaledToFit()
+                .frame(width: 24, height: 24)
+
+            Text(appFetcher.appName)
+
+            Spacer()
+        }
+        .frame(maxWidth: .infinity)
+        .contentShape(Rectangle())
+        .padding(.vertical, 4)
+        .padding(.leading, 6)
+        .task {
+            appFetcher.getAppBundleInfo()
+        }
+    }
+}
+
+@available(macOS 13.0, *)
+class AppFetcher: ObservableObject {
```

Comment:
I'm not sure if we really need a observable object here. Maybe we can just nested `getAppBundleInfo()` inside the initialization of `BlockAppItemView`?

Reply:
Yes, removed in [ceb5e36](https://github.com/tisfeng/Easydict/pull/340/commits/ceb5e3648b6b111dc26ffa636544dcd86c74a003),  looks more clear.